### PR TITLE
feat(layouts): add @mixin for responsive support for rows

### LINKED ISSUE #9112: md-divider: Responsive layout support is not properly supported
**Actual Behavior**:
-  `What is the issue? *` Divider will display as a column if the any responsive configuration of layout is set to `row`.
-  `What is the expected behavior?`Divider should only display as a column if the current layout is in a row configuration.

**CodePen** (or steps to reproduce the issue): *
-  `CodePen Demo which shows your issue:` http://codepen.io/shortfuse/pen/rLdEyR
-  `Details:` The problem is `divider.scss` is based around the `layout-row-*` only existing if it's displayed. The problem is the classes don't disappear when layout changes, locking divider into act as if it's always in a row configuration

**Angular Versions**: *
-  `Angular Version:` 1.1.0-rc5 (master)
-  `Angular Material Version:` 1.5.5

**Additional Information**:
-  `Browser Type: *` Chrome
-  `Browser Version: *` v52
-  `OS: *`ChromeOS

---

### PR DESCRIPTION
**Pull request by clshortfuse**
Create @mixin that allows different layout breakpoints

Fixes #9112

### CODE REVIEW

**Comment by clshortfuse** [comment]
http://codepen.io/shortfuse/pen/AXyZLX (localhost test)

File: src/core/style/mixins.scss [hunk 1]
```diff
@@ -326,3 +326,47 @@
     }
   }
 }
+
+// Only use when in row layout
+@mixin when-layout-row($element) {
```
Comment (crisbeto) [inline src/core/style/mixins.scss]: This seems like a whole lot of selectors (close 2kb), for something as simple as drawing a divider.

Comment (clshortfuse) [inline src/core/style/mixins.scss]: Right, I do have some PRs that would use this to fix issues with md-select, md-checkbox, md-input-container and md-datepicker. I had a tough time trying to squish it down though.

**Comment by clshortfuse** [comment]
I realized I should expand this to `divider-theme.scss` and `radio-button.scss` to make better use of the CSS

**Comment by clshortfuse** [comment]
Yeah, this is looking too bloated. Bumped radio-button from 1KB to 10KB.

File: src/core/style/mixins.scss [hunk 2]
```diff
@@ -326,3 +326,47 @@
     }
   }
 }
+
+// Only use when in row layout
+@mixin when-layout-row($element, $parent: '') {
+  @media (max-width: 599px) {
```
Comment (EladBezalel) [inline src/core/style/mixins.scss]: we have variables for those breakpoints

File: src/core/style/mixins.scss [hunk 3]
```diff
@@ -326,3 +326,47 @@
     }
   }
 }
+
+// Only use when in row layout
+@mixin when-layout-row($element, $parent: '') {
```
Comment (EladBezalel) [inline src/core/style/mixins.scss]: why not use the scss hierarchy instead of specify a parent?

**Comment by ThomasBurleson** [comment]
Mistakenly merged... just now reverted from `master`.

**Comment by ThomasBurleson** [comment]
Rejected as invalid PR

File: src/core/style/mixins.scss (hunk 3 above)
Comment (clshortfuse) [inline src/core/style/mixins.scss]: I've tried that, but the problem is that @media needs to be root level.

If I do ```
md-radio-group {
  @include when-layout-row(md-radio-button) {
    ...
  }
}```

Then the @media queries would appear under `md-radio-group`, I wasn't sure of a better solution.

**Pull request closed without merging**